content/docs/concepts/architecture.md

```diff
@@ -92,19 +90,23 @@ to 'refresh' the {{<color "#6A00FF" >}}vWire{{< /color >}}.  If a {{<color "#6A0
 
 A {{<color "#0050EF" >}}Client{{< /color >}} may use a 'MonitorConnection' streaming GRPC call to NSM to get updates on the status of a {{<color "#6A00FF" >}}vWire{{< /color >}} it has to a Network Service.
 
-## Registries
+## Registry
+
+The Network Service Registry (NSR) component tracks Network Services and Network Service {{<color "#008A00" >}}Endpoints{{</color>}}. It can use either in-memory volatile storage or Kubernetes custom resource storage.
```

Comment:
So first: thank you.  This is a vast improvement.

You have correctly described the existing registry implementations.  I'm not sure if this is the right place or not, but it may be worth noting that one can fairly trivially write additional implementations with the backing store of your choice, because the 'Registry API' is *just* a GRPC API.

Reply:
I pushed a commit that adds links to the gRPC definition and a few other tweaks.